### PR TITLE
refactor test & add codecov

### PR DESCRIPTION
**Pull request by cglotr**
- Refactor test to generate coverage reports
- Add workflow for codecov

### CODE REVIEW

File: .github/workflows/coverage.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    - run: docker-compose -f ./docker-compose.yml up --detach redis
+    - uses: actions/setup-python@v3
+      with:
+        python-version: "3.10.x"
+    - run: source ./activate.sh
+    - run: pip install --upgrade pip
+    - run: pip install -e .
+    - run: pip freeze
+    - run: ./test.sh
+    - uses: codecov/codecov-action@v3.1.0
```
Comment (cglotr) [inline .github/workflows/coverage.yaml:24]: Need to change this token since this belongs to the fork `cglotr/aiotaskq`.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # aiotaskq
 
+[![codecov](https://codecov.io/gh/cglotr/aiotaskq/branch/main/graph/badge.svg?token=9I3i2Pwtif)](https://codecov.io/gh/cglotr/aiotaskq)
```
Comment (cglotr) [inline README.md:3]: Need to change for `imranariffin/aiotaskq`.

File: .github/workflows/coverage.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      with:
+        token: e1038ca3-70ae-433d-be18-5764f119c744
```
Comment (cglotr) [inline .github/workflows/coverage.yaml:26]: Need to change this token since this belongs to the fork `cglotr/aiotaskq`.

Comment (cglotr) [reply .github/workflows/coverage.yaml:26]: file:///home/cglotr/Pictures/Screenshots/Screenshot%20from%202022-06-10%2004-25-11.png

File: .github/workflows/coverage.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    - run: docker-compose -f ./docker-compose.yml up --detach redis
+    - uses: actions/setup-python@v3
+      with:
+        python-version: "3.10.x"
+    - run: source ./activate.sh
+    - run: pip install --upgrade pip
+    - run: pip install -e .
+    - run: pip freeze
+    - run: ./test.sh
+    - uses: codecov/codecov-action@v3.1.0
```
Comment (cglotr) [inline .github/workflows/coverage.yaml:24]: Also need to add `CODECOV_TOKEN` in Github secret

Comment (imranariffin) [reply .github/workflows/coverage.yaml:24]: Is it this token? It says "Token not required for public repositories uploading from [Travis](https://travis-ci.org/), [CircleCI](https://circleci.com/), [AppVeyor](https://www.appveyor.com/), [Azure Pipelines](https://azure.microsoft.com/en-us/services/devops/pipelines/) or [GitHub Actions](https://help.github.com/en/actions)" so I'm wondering if we I really need to add the token.
Screenshot from 2022-06-19 07-00-43

**Review by imranariffin — CHANGES_REQUESTED** [review]
Looks good. Just a few minor changes related to links, and a question regarding codecov token.

File: .github/workflows/coverage.yaml (hunk 4 above)
Comment (cglotr) [reply .github/workflows/coverage.yaml:24]: Ok, not really sure as well but I guess no token needed for this

**Comment by cglotr** [comment]
@imranariffin removed tokens & fix readme badge paths. can help to merge this?

**Review by imranariffin — APPROVED** [review]

**Pull request merged**